Source/Core/SwipeActions.swift

```diff
@@ -0,0 +1,168 @@
+		}
+	}
+	
+	@available(iOS 11, *)
+	public var image: UIImage?{
+		get{
+			guard let contextualAction = self.contextualAction as? UIContextualAction else{ return nil }
+			return contextualAction.image
+		}
+		set{
+			guard let contextualAction = self.contextualAction as? UIContextualAction else{ return }
+			return contextualAction.image = newValue
+		}
+	}
+	
+	public var title: String?{
+		get{
+			if #available(iOS 11, *), let contextualAction = self.contextualAction as? UIContextualAction{
+				return contextualAction.title
+			
+			} else if let contextualAction = self.contextualAction as? UITableViewRowAction{
+				return contextualAction.title
+			}
+			return nil
+		}
+		set{
+			if #available(iOS 11, *), let contextualAction = self.contextualAction as? UIContextualAction{
+				contextualAction.title = newValue
+			
+			} else if let contextualAction = self.contextualAction as? UITableViewRowAction{
+				contextualAction.title = newValue
+			}
+		}
+	}
+	
+	public let handler: SwipeActionHandler
+	public let style: Style
+	
+	public init(style: Style, title: String?, handler: @escaping SwipeActionHandler){
+		self.style = style
+		self.handler = handler
+		
+		if #available(iOS 11, *){
+			self.contextualAction = UIContextualAction(style: style.contextualStyle as! UIContextualAction.Style, title: title){ action, view, completion -> Void in
+				handler(action, view, completion)
+			}
+			
+		} else {
+			self.contextualAction = UITableViewRowAction(style: style.contextualStyle as! UITableViewRowActionStyle,title: title){ (action, indexPath) -> Void in
+				handler(action, indexPath, nil)
+			}
+		}
+	}
+	
+	public enum Style: Int{
+		case normal = 0
+		case destructive = 1
+		
+		var contextualStyle: ContextualStyle{
+			if #available(iOS 11, *){
+				switch self{
+				case .normal:
+					return UIContextualAction.Style.normal
+				case .destructive:
+					return UIContextualAction.Style.destructive
+				}
+				
+			} else {
+				switch self{
+				case .normal:
+					return UITableViewRowActionStyle.normal
+				case .destructive:
+					return UITableViewRowActionStyle.destructive
+				}
+			}
+		}
+		
+		public init(contextualStyle: ContextualStyle){
+			if #available(iOS 11, *), let contextualStyle = contextualStyle as? UIContextualAction.Style{
+				switch contextualStyle{
+				case .normal:
+					self = .normal
+				case .destructive:
+					self = .destructive
+				}
+				
+			} else if let contextualStyle = contextualStyle as? UITableViewRowActionStyle{
+				switch contextualStyle{
+				case .normal:
+					self = .normal
+				case .destructive:
+					self = .destructive
+				case .default:
+					fatalError("Unmapped UITableViewRowActionStyle for SwipeActionStyle: \(contextualStyle)")
+				}
+				
+			} else {
+				fatalError("Invalid platformValue for SwipeActionStyle: \(contextualStyle)")
+			}
+		}
+	}
+}
+
+
+public class SwipeConfiguration{
+	
+	public init(configure: (SwipeConfiguration) -> Void){
```

Comment:
I would prefer a struct here and each row should have an instance of the strict already created so Eureka users can do something like that:

```
row.leftSwipe.actions = [swipeAction1, swipeAction2]
row.rightSwipe.actions = [swipeAction1, swipeAction2]
```

init should be refactored to `init()` in this case.

Reply:
This looks amazing. One question: Would you prefer using `leftSwipe` and `rightSwipe` instead of `leadingSwipe` and `trailingSwipe` ?

If I'll look at `PushRow`, `left/right` looks more familiar. But due to the nature of this gesture it might lead to confusions.

Reply:
Done using `leadingSwipe` and `trailingSwipe`.